Added Search methods for UmbracoHelper.ContentQuery (#2306)
umbraco/UmbracoDocs: Reference/Querying/UmbracoHelper/index.md

<p>@Umbraco.GetDictionaryValue("createdOn", "Date Created"): @Model.CreateDate</p>
```

### .Search(string term)

Given a search term, it by default searches the Umbraco search index for content matching the term. 

```csharp
@{
    <ul>
        @foreach (var result in Umbraco.ContentQuery.Search("ipsum"))
        {
            <li><a href="@result.Content.Url">@result.Content.Name</a></li>
        }
    </ul>
}
```

### .Search(string term, int skip, int take, out long totalRecords)

> You can also specify the number of records to skip, the number of records to take.

Reviewer: should we say this is more performant, when you want to implement 'paging' ?

```csharp
@{
    var search = Umbraco.ContentQuery.Search("ipsum", 5, 10, out long totalRecords);
    <ul>
        <li>
            Total results: @totalRecords
            <ul>
                @foreach (var result in search)
                {
                    <li><a href="@result.Content.Url">@result.Content.Name</a></li>
                }
            </ul>
        </li>
    </ul>
}
```

### .Search(IQueryExecutor queryExecutor)

> For more complex searching you can construct an Examine QueryExecutor

Reviewer: Should this example be more complex?

```csharp
@{
    if (!ExamineManager.Instance.TryGetIndex(Constants.UmbracoIndexes.ExternalIndexName, out IIndex index))
    {
        throw new InvalidOperationException($"No index found by name{ Constants.UmbracoIndexes.ExternalIndexName }");
    }

    var term = "yellow";
    var query = index.GetSearcher().CreateQuery(IndexTypes.Content);

>     var queryExecutor = query.ManagedQuery(term);

Reviewer: should we add some kind of additional complexity to the filtering here? by section? or of certain doc type?

<p>@Umbraco.GetDictionaryValue("createdOn", "Date Created"): @Model.CreateDate</p>
```

### .Search(string term)

> Given a search term, it by default searches the Umbraco search index for content matching the term. 

Reviewer: ```suggestion
By default, Umbraco searches it's 'External' search index for any published content matching the provided search term. 
```

```csharp
@{
    <ul>
        @foreach (var result in Umbraco.ContentQuery.Search("ipsum"))
        {
            <li><a href="@result.Content.Url">@result.Content.Name</a></li>
        }
    </ul>
}
```

### .Search(string term, int skip, int take, out long totalRecords)

> You can also specify the number of records to skip, the number of records to take. This is more performant when have a lot of search results and need to implement paging.

Reviewer: ```suggestion
Specifying the number of records 'to skip', and the number of records 'to take' is more performant when there are lots of matching search results and there is a requirement to implement paging.
```

```csharp
@{
    var search = Umbraco.ContentQuery.Search("ipsum", 5, 10, out long totalRecords);
    <ul>
        <li>
            Total results: @totalRecords
            <ul>
                @foreach (var result in search)
                {
                    <li><a href="@result.Content.Url">@result.Content.Name</a></li>
                }
            </ul>
        </li>
    </ul>
}
```

### .Search(IQueryExecutor queryExecutor)

For more complex searching you can construct an Examine QueryExecutor. In the example the search will execute against content of type "blogPost" only.
[Further information on using Examine](../../Searching/Examine/Quick-Start/index.md#different-ways-to-query)

```csharp
@{
    if (!ExamineManager.Instance.TryGetIndex(Constants.UmbracoIndexes.ExternalIndexName, out IIndex index))
    {
        throw new InvalidOperationException($"No index found by name{ Constants.UmbracoIndexes.ExternalIndexName }");
    }

>     var term = "consectetur";

Reviewer: ```suggestion
    var term = "ipsum";
```
